fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -338,7 +338,11 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
         when (getPreviousDestination()?.destination?.id) {
             R.id.browserFragment, R.id.homeFragment -> {
                 binding.searchWrapper.setOnTouchListener { _, _ ->
-                    binding.searchWrapper.hideKeyboard()
+                    if (toolbarView.view.url.isEmpty()) {
+                        dismissAllowingStateLoss()
```

Comment:
I think adding the dismiss call here breaks the interaction with the home fragment private [button](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt#273): 

https://user-images.githubusercontent.com/35462038/234011207-c43d6a22-1db3-4f70-adea-b06cbd4ba322.mp4